### PR TITLE
feat(publish-release): publishing releases to gh-pages

### PR DESCRIPTION
**Pull request by sunix**
### What does this PR do?
When releasing, publishing che-plugin-registry to github pages

### What issues does this PR fix or reference?
Fix https://github.com/eclipse/che/issues/19268

### How to test this PR?
I have tested this on my fork with a copy of release.yml without the release steps: https://github.com/sunix/che-plugin-registry/commit/f35cf908685aeebdb248ef10d922045c0f22d767

And triggered manually with versions as input:
the results:
- https://sunix.github.io/che-plugin-registry/7.27.0/v3/plugins/
- https://sunix.github.io/che-plugin-registry/7.28.1/v3/plugins/
- https://sunix.github.io/che-plugin-registry/7.28.2/v3/plugins/

Also how the git history would look like in the gh-pages branch: https://github.com/sunix/che-plugin-registry/commits/gh-pages
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: build/dev/publish-devfile-registry-to-gh-pages.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+set -e
+
+DEFAULT_BUILD_DIR="/projects/che-plugin-registry/output"
+BUILD_DIR=${BUILD_DIR:-$DEFAULT_BUILD_DIR}
+
+DEFAULT_GH_PAGES_DIR="/projects/gh-pages"
+GH_PAGES_DIR=${GH_PAGES_DIR:-$DEFAULT_GH_PAGES_DIR}
+
+# not setting eclipse-che in purpose to avoid errors committers pushing in the main registry
+DEFAULT_GITHUB_ACTOR="sunix"
```
Comment (benoitf) [inline build/dev/publish-devfile-registry-to-gh-pages.sh:18]: default should not be sunix

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -90,6 +90,21 @@ jobs:
           export GITHUB_TOKEN=${{ secrets.CHE_BOT_GITHUB_TOKEN }}
 
           ./make-release.sh --version ${{ github.event.inputs.version}} --trigger-release
+      - name: Extract content for publishing
```
Comment (ericwill) [inline .github/workflows/release.yml:93]: Do we want to also add nightly (on push) publish to GitHub pages? Or GitHub pages = released version, nightly = surge.sh?

Comment (sunix) [reply .github/workflows/release.yml:93]: yes we could do that. maybe in a second  PR? also planning to improve the dogfooding devfile using the same script.

File: build/dev/publish-devfile-registry-to-gh-pages.sh (hunk 1 above)
Comment (sunix) [reply build/dev/publish-devfile-registry-to-gh-pages.sh]: I finally don't set any default value: the script will check and fail if GITHUB_ACTOR and GITHUB_TOKEN are not set

File: .github/workflows/release.yml (hunk 2 above)
Comment (ericwill) [reply .github/workflows/release.yml:93]: :+1:

**Review by ericwill — APPROVED** [review]
Works for me, it seems the correct content is there.

**Pull request merged**